tests/test_kea.py

```diff
@@ -61,12 +63,33 @@ def test_kea_dhcp4(
             cli_launcher.launch_container()
 
             cli_con = cli_launcher.container_data.connection
+            # get the default interface name
             default_interface = cli_con.check_output(
-                "ip route show | grep default | cut -d' ' -f5"
+                "ip link show up | grep -v \"lo\" | head -n 1 | cut -d':' -f2 | cut -d'@' -f1 | tr -d ' '"
```

Comment:
this seems really fragile.  for example for me the "lo" device does not have double quotes around it (locale issue)?

I would suggest to use the json output, e.g. something like

`f"ip -4 -j route get {request_ip} | jq -r '.[0].dev'"`

jq might not be available, but you could use python's json library to parse it.